### ISSUE TITLE
When RegisterSingle is called on a PocketContainer clone, it registers the singleton in the container from which the clone was created

### ISSUE THREAD

**Issue by jonsequitur** [jonsequitur/PocketContainer#16]
The following fails:

```csharp
            var original = new PocketContainer();
            original.RegisterSingle(c => new object());

            var clone = original.Clone();
            clone.RegisterSingle(c => new object());

            var resolvedFromClone = clone.Resolve<object>();
            var resolvedFromOriginal = original.Resolve<object>();

            resolvedFromOriginal.Should()
                                .NotBeSameAs(resolvedFromClone);
```

**Comment by TyOverby** [comment]
Looks like the lack of a clone for singletons

https://github.com/jonsequitur/PocketContainer/blob/9e5952a2c6280bea086ed4144ea842da9f64f60b/Pocket.Container/PocketContainer.Clone.cs#L20

**Comment by jonsequitur** [comment]
This was deliberate in order to support Microsoft.Extensions.DependencyInjection scope-based disposal but I'm looking for another way forward on that and may make this change.

**Comment by jonsequitur** [comment]
Fixed by #33.
